OpenRA.Mods.Common/Activities/GenericDockSequence.cs

```diff
@@ -24,37 +24,37 @@ public class GenericDockSequence : Activity
 	{
 		protected enum DockingState { Wait, Drag, Dock, Loop, Undock, Complete }
 
-		protected readonly Actor DockHostActor;
-		protected readonly IDockHost DockHost;
-		protected readonly WithDockingOverlay DockHostSpriteOverlay;
-		protected readonly DockClientManager DockClient;
-		protected readonly IDockClientBody DockClientBody;
+		protected readonly Actor LinkHostActor;
+		protected readonly ILinkHost LinkHost;
+		protected readonly WithDockingOverlay LinkHostSpriteOverlay;
```

Comment:
Renaming variables from Dock to Link doesn't feel right - this activity (and the `DockHost` trait are the concrete implementation of the docking behaviour, so it would be clearer if the variable names reflected that.

The way I picture it is that there is the base "Link" concept, of which `ProximityLink` and `DockHost` are concrete implementations. Things that don't care about proximity-vs-dock should be using the general Link terminology, but things that do care should be picking a side so that its immediately clear to the reader what the expectations are.

Reply:
unnamed